process/tags.go

```diff
@@ -0,0 +1,119 @@
+package process
+
+import (
+	"encoding/binary"
+	"math"
+)
+
+type TagEncoder interface {
```

Comment:
Is it worth adding a Reset() operation that clears the buffer so we can re-use this?

Reply:
No, I don't think so.

The underlying buffer will used directly by the `EncodedTags` field in the protobuf so there really isn't a good way to reuse it at the moment.